draft-barnes-acme.md

```diff
@@ -516,18 +544,211 @@ Link: <https://example.com/acme/terms>;rel="terms-of-service"
+* The MAC verifies using the recovery key
+* The JWK in the payload represents the new account key (i.e. the key used to verify the ACME message)
+
+If those conditions are met, and the recovery request is otherwise acceptable to the server, then the recovery process has succeeded.  The server creates a new registration resource based on the base registration and the new account key, and returns it on a 201 (Created) response, together with a Location header indicating a URI for the new registration.  If the recovery request is unsuccessful, the server returns an error response, such as 403 (Forbidden).
+
+~~~~~~~~~~
 
+HTTP/1.1 201 Created
+Content-Type: application/json
+Location: https://example.com/reg/asdf
+Link: <https://example.com/acme/new-authz>;rel="next"
+Link: <https://example.com/acme/recover-reg>;rel="recover"
+Link: <https://example.com/acme/terms>;rel="terms-of-service"
+
+{
+  "key": { /* JWK from JWS header */ },
 
-### Account Recovery
+  "contact": [
+    "mailto:cert-admin@example.com",
+    "tel:+12025551212"
+  ],
+
+  "authorizations": [...],
+  "certificate": [...]
+}
+
+~~~~~~~~~~
+
+
+
+### Contact-Based Recovery
+
+In the contact-based recovery process, the client requests that the server send a message to one of the contact URIs registered for the account.  That message indicates some action that the server requires the client's user to perform, e.g., clicking a link in an email.  If the user successfully completes the server's required actions, then the server will bind the account to the new account key.
+
+(Note that this process is almost entirely out of band with respect to ACME.  ACME only allows the client to initiate the process, and the server to indicate the result.)
+
+To initiate contact-based recovery, the client sends a POST request to the server's recover-registration URI, with a body specifying which registration is to be recovered.  The body of the request MUST be signed by the client's new account key pair.
+
+method (required, string):
+: The string "contact"
+
+base (required, string):
+: The URI for the registration to be recovered.
+
+~~~~~~~~~~
+
+POST /acme/recover-registration HTTP/1.1
+Host: example.com
+
+{
+  "resource": "recover-reg",
+  "method": "contact",
+  "base": "https://example.com/acme/reg/asdf"
+}
+/* Signed as JWS, with new account key */
+
+~~~~~~~~~~
+
+If the server agrees to attempt contact-based recovery, then it creates a new registration resource containing a stub registration object.  The stub registration has the client's new account key and anonymized contacts, in order to allow the the client to know which contacts to check.  The server returns the stub contact in a 201 (Created) response, along with a Location header field indicating the URI for the new registration resource (which will be the registration URI if the recovery succeeds).
+
+~~~~~~~~~~
+
+HTTP/1.1 201 Created
+Content-Type: application/json
+Location: https://example.com/reg/qwer
```

Comment:
It doesn't actually matter, because the URLs are examples.  But I fixed the other ones.